tests/bindings/python/oval_helpers.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+
+# Author:
+#   Dominique Blaze <contact@d0m.tech>
+#
+# Description:
+#   Helpers functions in order to factorize other tests features
+#   It can potentially be integrated later in the high layer python api
+#   to provide convenience utilities.
+#
+
+
+def browse_criteria(crit_node, mode=0):
+    '''
+    Browse recursively criteria of an oval test and build a representation of it
+    First list item is the operator, and the next ones are the tests
+    If the item is a list, it's a criteria (else a criterion, or string for an
+    extended defintion , or None for OVAL_NODETYPE_UNKNOWN
+
+    Ex: [operator, "extended def", [operator, test_result2, test_result3]]
+
+    Mode : with mode = 0, test_result is the swig object oval_test_result
+    with mode = 1, test result is a a tuple like (test_subtype, test_result)
+    for instance (oscap.oval.OVAL_LINUX_DPKG_INFO, OVAL_FAMILY_LINUX
+    oscap.xccdf.XCCDF_RESULT_PASS)
+    '''
+
+    # init the critria list
+    rs = list()
+
+    if crit_node.get_type() == oscap.oval.OVAL_NODETYPE_CRITERIA:
```

Comment:
I have the following traceback in `ctest --verbose -R bindings/test_python.sh`:

```
109: 
109: + python_oval_eval                                             
109: TEST: python_oval_eval
109: /home/jcerny/openscap/build/run python3 /home/jcerny/openscap/tests/bindings/python/oval_eval.py
109: Traceback (most recent call last):
109:   File "/home/jcerny/openscap/build/swig/python3/openscap_api.py", line 391, in __output_callback
109:     return obj[0](OSCAP_Object(extract_type_from_obj(result), result), obj[1])
109:   File "/home/jcerny/openscap/tests/bindings/python/oval_eval.py", line 52, in oval_sample_callback
109:     tests_tree = browse_criteria(ovdef.get_criteria(), 1)
109:   File "/home/jcerny/openscap/tests/bindings/python/oval_helpers.py", line 31, in browse_criteria
109:     if crit_node.get_type() == oscap.oval.OVAL_NODETYPE_CRITERIA:
109: NameError: name 'oscap' is not defined
109: openscap_api loaded from /home/jcerny/openscap/build/swig/python3/openscap_api.py
109: Opening oval file with original C functions ...  
109: oval:foo_pass:def:1 => PASS
109: RESULT: PASSED
```

Could you take a look into this?

Also I find suspicious that the test result is PASS when there was a traceback.

Reply:
@jan-cerny  Yep nice catch, in fact the error raised into the callback function aren't handled by the main script ... i'm working to correct this.